packages/app-builder-lib/src/macPackager.ts

```diff
@@ -192,6 +192,15 @@ export default class MacPackager extends PlatformPackager<MacConfiguration> {
       }
     }
 
+    async function processTargetsPack(targets: Array<Target>, index: number) {
+      if (index < targets.length) {
+        await processTargetPack(targets[index]);
+        await processTargetsPack(targets, index + 1);
```

Comment:
I love that you extracted the logic to a single `processTargetPack`

This logic should fail if there are more than two targets specified though, such as `"target":  ["mas", "dmg", "zip"],`, right?

How is this recursion different than the current loop of `for (const target of targets)`?

I don't see anything different here from this code
```
for (const target of targets) {
      await processTargetPack(target)
}
```

Reply:
```
for (const target of targets) {
      await processTargetPack(target)
}
```
This code causes the asynchronous tasks inside the loop to fire at the same time, rather than one after the other

Reply:
Oh, sorry, I made a mistake when debugging, which made me think I found the real cause of this problem. Now I have resubmitted my edits, please check